src/cryptonote_basic/tx_extra.h

```diff
@@ -425,6 +425,20 @@ namespace cryptonote
     timesync_status_out_of_sync = 1 << 5
   };
 
+  inline const char* ToString(OS_type v)
+  {
+      switch (v)
+      {
+          case missed_uptime_proof:   return "Missed Uptime Proofs";
+          case missed_checkpoints:   return "Missed Checkpoints";
+          case missed_pulse_participations: return "Missed Pulse Participation";
+          case storage_server_unreachable: return "Storage Server Unreachable";
+          case timestamp_response_unreachable: return "Unreachable for Timestamp Check";
+          case timesync_status_out_of_sync: return "Time out of sync";
+          default:      return "unknown decommission reason";
+      }
+  }
```

Comment:
Instead of making this a named method you can do:
```c++
inline std::ostream& operator <<(std::ostream& o, OS_type v) {
  switch (v)
  {
    case missed_uptime_proof: return o << "Missed Uptime Proofs";
    // etc.
  }
}
```
which makes it directly printable in an output stream as `blah << reason` instead of `blah << ToString(reason)`.